src/dtproject/types.py

```diff
@@ -158,5 +159,26 @@ class DevContainerConfiguration(dict):
     workspaceFolder: Optional[str] = None
 
 
+@dataclasses.dataclass
+class Hook:
+    command: str
+    required: Optional[bool] = False
+
+
+@dataclasses.dataclass
+class LayerHooks(DataClassLayer):
+    hooks: Dict[str, List[Hook]] = dataclasses.field(default_factory=lambda: ddict(list))
+
+    def __iter__(self) -> Iterator[Tuple[EventName, List[Hook]]]:
+        for e, h in self.hooks.items():
+            yield e, h
+
+    def __getitem__(self, item):
+        return self.hooks[item]
+
+    def get(self, __key, _=None) -> list:
+        return self.hooks[__key]
+
```

Comment:
I modified this otherwise when using hooks loaded from the yaml file and trying to acces a non-existant key we would receive `KeyError`.